pkg/provider/aws/static_unit.go

```diff
@@ -16,6 +16,8 @@ func (a *AWS) GenerateStaticUnitResources(unit *core.StaticUnit, dag *core.Resou
 		object := resources.NewS3Object(bucket, filepath.Base(f.Path()), f.Path(), filepath.Join(unit.ID, f.Path()))
 		dag.AddResource(object)
 		dag.AddDependency2(object, bucket)
+		a.MapResourceDirectlyToConstruct(object, unit)
 	}
+	a.MapResourceDirectlyToConstruct(bucket, unit)
```

Comment:
hm, _does_ the bucket come directly from the static unit, though? It feels like either the bucket or the objects should, but not both.

I'm not really sure about that, though. It's a bit strange: the annotation is file/object-based, but the yaml.config is bucket-based (ish).

To ground my question a bit more, I guess I'll ask: what do we need these for? Is it just general graph hygiene, or a more concrete issue it's addressing?

Reply:
yeah right now each static unit generates their own bucket and the objects from it. we need the bucket for wiring up cloudfront. 

Im a little confused on the questions here to be honest if you dont mind elaborating. Static unit is the bucket and the contents within that bucket

Reply:
My question is about the "directly" aspect. I can think of three ways to think of how the bucket and the objects relate to the construct:

1. The construct directly implies the objects; the bucket is there as a supporting resource for those objects
2. The construct directly implies the bucket; the objects are there as an implementation detail of the bucket
3. The construct directly implies both the objects and the bucket.

Of those, this code is option 3. But options 1 and 2 make sense to me, too.

(Option 1 has the most UX impact: it means that if the user wrote a `@klotho::static_unit` that ended up including zero files, that we would then elide away the bucket, since it's no longer needed to support the non-existent objects.)

The DAG will already say that the objects depend on the bucket, so we'll be able to get from the construct to the objects regardless of the approach. My only question is the "directly" part, and I'm only really asking it because you mentioned that we should be really careful about which resources we map directly to constructs.

What I meant by this:

> To ground my question a bit more, I guess I'll ask: what do we need these for? Is it just general graph hygiene, or a more concrete issue it's addressing?

... is to ask basically what would happen if we didn't have these `MapResourceDirectlyToConstruct` calls, from a UX perspective. That might help us pick which of the three options we want.

Again, I'm only asking because you had warned against being too liberal with these calls. If it's not a problem to include them, I'm fine with it. Seems like a 2-way door either way, right?

Reply:
yeah seems like a 2 way door, i see your point. In my opinion the annotation denotes the bucket and the objects are a description of the contents within the bucket. We really care mainly about the bucket in the case of things being mapped since thats what we need to retrieve for configuration on the static unit annotation itself. Im fine removing it, the reason i did both is because those objects can only ever belong to the single static unit annotation as well, so to me they are also a direct mapping to the annotation

Reply:
>  those objects can only ever belong to the single static unit annotation as well

by that reasoning, a lambda's execution role would be directly mapped to the `@klotho::execution_unit` annotation, no?

Reply:
yeah i think thats reasonable